python/cog/server/worker.py

```diff
@@ -417,11 +420,32 @@ def _current_tag(self) -> Optional[str]:
             return tag
         return self._sync_tag
 
-    def _setup(self, redirector: AsyncStreamRedirector) -> None:
+    def _assign_predictor(self) -> None:
         done = Done()
         wait_for_env()
         try:
             self._predictor = load_predictor_from_ref(self._predictor_ref)
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            traceback.print_exc()
+            done.error = True
+            done.error_detail = str(e)
+        except BaseException as e:
+            # For SystemExit and friends we attempt to add some useful context
+            # to the logs, but reraise to ensure the process dies.
+            traceback.print_exc()
+            done.error = True
+            done.error_detail = str(e)
+            raise
+        finally:
+            self._events.send(Envelope(event=done))
```

Comment:
this is a bug. we send a `done` when setup is complete, but we don't send one when we finish importing the predictor